packages/lesswrong/components/thinkPage/ThinkLink.tsx

```diff
@@ -0,0 +1,58 @@
+// TODO: Import component in components.ts
+import React from 'react';
+import { registerComponent, Components } from '../../lib/vulcan-lib';
+import { useTracking } from "../../lib/analyticsEvents";
+import { Link } from '@/lib/reactRouterWrapper';
+
+const styles = (theme: ThemeType) => ({
+  root: {
+    ...theme.typography.body2,
+    color: theme.palette.grey[600]
+  }
+});
+
+export const getThinkPostBaseUrl = (document: PostsListWithVotes) => {
+  return `/think/posts/${document._id}/${document.slug}`
+}
+
+export const getThinkPostUrl = (document: PostsListWithVotes, forceEdit?: boolean) => {
+  const edit = (document.draft || forceEdit) ? `?edit=true&key=${document.linkSharingKey}` : ''
+  return `${getThinkPostBaseUrl(document)}${edit}`
+}
+
+export const getThinkSequenceUrl = (document: SequencesPageTitleFragment, forceEdit?: boolean) => {
+  const edit = (document.draft || forceEdit) ? '?edit=true' : ''
+  return `/think/seq/${document._id}${edit}`
```

Comment:
I think I weakly prefer retaining consistency with existing convention for `/s/${sequenceId}/p/${postId}` links